File: 2021/Days/Day1-2.py
with open('Python\\Advent-Of-Code\\2021\\Input\\Day1input.txt') as f:
    lines = list(map(int, f.readlines()))

c = 0
for p in range(3, len(lines)):
    if (sum(lines[p-3:p]) < sum(lines[p-2:p+1])):
        c += 1

print(c)

File: 2022/advent/src/bin/day1-2.rs
use advent::lib;
use std::cmp;

fn main(){
    let mut max: Vec::<u32> = vec!();
    let mut sum:u32 = 0;
    if let Ok(lines) = lib::read_lines("./in/day1input.txt"){
        for line in lines {
            if let Ok(ip) = line {
                if ip == ""{
                    max.push(sum);
                    sum = 0;
                }else{
                    sum += ip.parse::<u32>().unwrap();
                }
            }
        }
    }
    max.sort();
    let vec_len = max.len();
    println!("{}", max[vec_len-1] + max[vec_len-2] + max[vec_len-3]);
}